### PR TITLE
[develop] Fix inconsistency with berkeley circuit

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR fixes the inconsistency between the circuits generated by pickles in `berkeley` and those in `develop`. This PR also includes a test that the checked-in file matches the vk generated by the version of the code in a PR.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by nholland94 — APPROVED** [review]
LGTM, but can you add some output to the unit-test script that explains what to do when the verification key diff fails?

File: src/app/print_blockchain_snark_vk/dune [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ (deps print_blockchain_snark_vk.exe)
+ (targets blockchain_snark_vk.json.corrected)
+ (action
+  (with-stdout-to %{targets}
+   (run %{deps}))))
+
+(rule
+ (alias runtest)
+ (action
+  (diff blockchain_snark_vk.json blockchain_snark_vk.json.corrected)))
```
Comment (nholland94) [inline src/app/print_blockchain_snark_vk/dune:18]: Maybe this should be piped to `/dev/null`? Or does `dune runtest` already suppress the output here?

**Review by dannywillems — APPROVED** [review]
I think the dune-project file and ocamlformat are not necessary.

**Comment by mrmr1993** [comment]
> I think the dune-project file and ocamlformat are not necessary.

The dune-project file is because disabling transitive implicit deps annoys me, and I can turn it back on by making a new project. I think we agreed to remove it everywhere anyway, just not in this PR.

File: src/app/print_blockchain_snark_vk/dune (hunk 1 above)
Comment (mrmr1993) [reply src/app/print_blockchain_snark_vk/dune:18]: It should already suppress it, it only goes into the file

**Comment by mrmr1993** [comment]
> can you add some output to the unit-test script that explains what to do when the verification key diff fails?

@nholland94 dune already shows the relevant information in the error message: it shows a diff, and tells you that you can run `dune promote` to update the generated file. Hopefully that's enough?

**Pull request merged**